en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml

```diff
@@ -0,0 +1,636 @@
+    scopes:
+      apim:api_publish: Publish API
+
+######################################################
+# The "API Collection" resource APIs
+######################################################
+paths:
+  /deploy-api:
+
+    #-----------------------------------------------------
+    # Deploy the API in the Gateway
+    #-----------------------------------------------------
+    post:
+      security:
+        - OAuth2Security:
+            - apim:api_publish
+      x-wso2-curl: "curl -k -H \"Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8\" -X POST \"https://127.0.0.1:9443/api/am/gateway/v1/deploy-api?apiId=890a4f4d-09eb-4877-a323-57f6ce2ed79b&apiName=PizzaAPI&label=Food\""
+      x-wso2-request: |
+        POST https://127.0.0.1:9443/api/am/gateway/v1/deploy-api?apiId=890a4f4d-09eb-4877-a323-57f6ce2ed79b&apiName=PizzaAPI&label=Food
+        Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8
+      x-wso2-response: "HTTP/1.1 200 OK"
+      summary: Deploy the API in the gateway
+      description: |
+        This operation is used to deploy and API in the gateway.
+      parameters:
+        - name: apiName
+          description: |
+            Name of the API
+          in: query
+          type: string
+          required: true
+        - name: label
+          description: |
+            Label of the API Gateway
+          in: query
+          type: string
+          required: true
+        - $ref: '#/parameters/apiId-Q'
+
+      tags:
+        - Deploy API
+      responses:
+        200:
+          description: |
+            OK.
+            API successfully deployed in the Gateway.
+          schema:
+            $ref: '#/definitions/DeployResponse'
+        202:
+          description: |
+            Accepted.
+            The request has been accepted.
+          schema:
+            $ref: '#/definitions/DeployResponse'
+        400:
+          description: |
+            Bad Request.
+            Invalid request or validation error
+          schema:
+            $ref: '#/definitions/Error'
+        404:
+          description: |
+            Not Found.
+            Requested API does not exist.
+          schema:
+            $ref: '#/definitions/Error'
+        412:
+          description: |
+            Precondition Failed.
+            The request has not been performed because one of the preconditions is not met.
+          schema:
+            $ref: '#/definitions/Error'
+
+  /undeploy-api:
+
+    #-----------------------------------------------------
+    # Undeploy the API in the Gateway
+    #-----------------------------------------------------
+    post:
+      security:
+        - OAuth2Security:
+            - apim:api_publish
+      x-wso2-curl: "curl -k -H \"Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8\" -X POST \"https://127.0.0.1:9443/api/am/gateway/v1/deploy-api?apiId=890a4f4d-09eb-4877-a323-57f6ce2ed79b&apiName=PizzaAPI&label=Food\""
```

Comment:
Don't we have to implement this with the same query params as GET api-artifact resource?  with tenant domain, API name and version.

Reply:
Fixed https://github.com/wso2/docs-apim/pull/1527/commits/96a86f56bf4414db5ee984eb37e500c62d467e54

Reply:
Fixed https://github.com/wso2/docs-apim/pull/1527/commits/2f6d4685e563ba2752305b94dc8bbc50a4335693